examples/threading/threading.c

```diff
@@ -0,0 +1,100 @@
+/********************************************************************************************************************************************************
+ File name: threading.c
+ Description: To create a thread that waits, obtains a mutex, holds it, and then releases it after certain time intervals 
+ Modified by: Vidhya. PL
+ Date : 09/20/2023
+ Credits : start_thread_obtaining_mutex() was partially generated using ChatGPT at https://chat.openai.com/ 
+ 	   with prompts including "Complete the TODO for this functions"
+ **********************************************************************************************************************************************************
+ */
+
+#include "threading.h"
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+// Optional: use these functions to add debug or error prints to your application
+#define DEBUG_LOG(msg,...)
+//#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
+#define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
+
+void* threadfunc(void* thread_param)
+{
+
+    // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
+    // hint: use a cast like the one below to obtain thread arguments from your parameter
+    //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+ 
+    //Casting the thread parameter back to the expected struct type. 	
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+
+    // Sleep for wait_to_obtain_ms milliseconds.
+    usleep(thread_func_args->wait_to_obtain_ms * 1000);
```

Comment:
@vidhyapalaniappan is it possible for `usleep` to fail to sleep the requested amount of time?